src/main/java/bio/terra/common/UriUtils.java

```diff
@@ -0,0 +1,70 @@
+package bio.terra.common;
+
+import autovalue.shaded.kotlin.text.Charsets;
+import bio.terra.service.filedata.exception.InvalidUriException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Objects;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.client.utils.URLEncodedUtils;
+
+/** Handy methods for working with URIs */
+public class UriUtils {
+
+  private UriUtils() {}
+
+  /**
+   * Extract the value from a query parameter
+   *
+   * @param uri a valid uri
+   * @param parameter the name of the parameter whose value to read
+   * @return the value of the query parameter or null. If there are multiple instances of the query
+   *     parameter, return the value as a comma-separated list
+   */
+  public static String getValueFromQueryParameter(String uri, String parameter) {
```

Comment:
In the sprit of YAGNI I'd like to point out that both `getValueFromQueryParameter()` and `omitQueryParameter()` are only invoked with `GcsConstants.USER_PROJECT_QUERY_PARAM` as a value for `parameter`. Is a general method needed here, or would one that is specific to the current use case be sufficient?

Reply:
I wanted a general method since this is something I can see us needing and I wanted to just have it out there for the next person